### ISSUE TITLE
Redirection Error

### ISSUE THREAD

**Issue by clementdroff** [johngodley/redirection#751]
Hello,

Thank you for developping this plugin! It is very useful.

Since a few days, I've experiencing problem when I want to create a new redirection. I try to deactivate WordFence but there is still the same problem.

Here are the details :  I appreciate a lot if you can help or point me the right direction!

```
Plugin: 3.1.1
WordPress: 4.9.4 (single)
PHP: 7.0.27
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.9; rv:58.0) Gecko/20100101 Firefox/58.0
REST API: https://www.bieresbretonnes.fr/wp-json/

Error: JSON.parse: unexpected character at line 1 column 1 of the JSON data
Action: redirection/v1/redirect POST
Params: {"id":0,"url":"https://www.bieresbretonnes.fr/brasseries/brasserie-malt-et-mouse/","title":"","regex":false,"match_type":"url","action_type":"url","position":0,"group_id":1,"action_code":301,"action_data":{"url":"https://www.bieresbretonnes.fr/brasseries/brasserie-malt-et-mouse/"}}
Code: 403 Forbidden
Raw: <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>403 Forbidden</title>
</head><body>
<h1>Forbidden</h1>
<p>You don't have permission to access /wp-json/redirection/v1/redirect
on this server.<br />
</p>
</body></html>

```

**Comment by Vleod** [comment]
I am getting the same issue but only on non-admin roles.

**Comment by johngodley** [comment]
Is this only happening when you create things?

**Comment by clementdroff** [comment]
Hi John!

Yes. The old redirections work fine.

It happens only when I create new redirections or when I try to create a new group or save new settings.

I deactivate and deinstall the plugin, I updated it with version 3.2. I also try to deactivate other plugin like Wordfence or WP Rocket.

Thanks!

**Comment by johngodley** [comment]
Are you running any kind of security software, or maybe an IP blocker?

**Comment by johngodley** [comment]
I am seeing this error for a few people, but for different reasons.

In one instance Varnish was removing cookies, and so the request wasn't authenticated. In another I found the server was rejecting `application/json` requests for some reason.

I have a new version 3.2.1 which I think works around the application/json issue.

https://github.com/johngodley/redirection/releases/tag/3.2.1

Can you give it a try and see if it helps? If you still get the error you should try the other 'REST API' settings in the error message.

**Comment by johngodley** [comment]
Closing for now, please reopen with more details

**Comment by clementdroff** [comment]
Hi!

Sorry for the delay. The website that have this problem is a side project for me.

Thank you for the answer. Yes I use security software but I tried several times to deactivate and did full uninstallation. Same thing for caching plugin WP Rocket.

I tested 3.2.1 beta and "Proxy over admin ajax" and "form request" parameters do work!

Thank you. I hope that help others having the same problem.

**Comment by clementdroff** [comment]
Hi!

Sorry for the delay. The website that have this problem is a side project for me.

Thank you for the answer. Yes I use security software but I tried several times to deactivate and did full uninstallation. Same thing for caching plugin WP Rocket.

I tested 3.2.1 beta and "Proxy over admin ajax" and "form request" parameters do work!

Thank you. I hope that help others having the same problem.
